src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,132 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] internalArray;
+    private int size;
+
+    public ArrayList() {
+        this.internalArray = new Object[DEFAULT_CAPACITY];
+    }
+
+    public ArrayList(int capacity) {
+        this.internalArray = new Object[capacity];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size < internalArray.length) {
+            internalArray[size] = value;
+            size++;
+        } else {
+            grow();
+            internalArray[size] = value;
+            size++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index >= 0 && index <= size) {
+            shiftToRight(index);
+            internalArray[index] = value;
+            size++;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index. The index "
+                    + "'" + index + "' doesn't exist");
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < size && index >= 0) {
+            return (T) this.internalArray[index];
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("The such index doesn't exist");
+        }
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index < size && index >= 0) {
+            internalArray[index] = value;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Wrong array index!");
+        }
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        T oldelement = null;
+        if (index < this.size && index >= 0) {
+            oldelement = get(index);
+            Object[] newFormedArray = new Object[internalArray.length];
+            System.arraycopy(internalArray, 0, newFormedArray, 0, index);
+            System.arraycopy(internalArray, index + 1, newFormedArray, index, (size - index) - 1);
+            internalArray = newFormedArray;
+            size--;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index! Index '"
+                    + index + "' doesn't exist");
+        }
+        return oldelement;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        T oldElement = null;
+        for (int i = 0; i < size; i++) {
+            if (internalArray[i] != null && internalArray[i].equals(element)) {
```

Comment:
no need for if-else if body is the same in both cases, combine into correct null safe comparison condition